test/e2e/storage/volumeattributesclass.go

```diff
@@ -0,0 +1,178 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	storagev1alpha1 "k8s.io/api/storage/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+var _ = utils.SIGDescribe("VolumeAttributesClass", func() {
+
+	f := framework.NewDefaultFramework("vac")
+	f.NamespacePodSecurityLevel = admissionapi.LevelBaseline
+
+	ginkgo.BeforeEach(func(ctx context.Context) {
+		e2eskipper.SkipIfMissingResource(ctx, f.DynamicClient, storagev1alpha1.SchemeGroupVersion.WithResource("volumeattributesclasses"), false /* namespaced */, "")
+	})
+
+	/*
+		Release: v1.29
+		Testname: VolumeAttributesClass, lifecycle
+		Description: Creating a VolumeAttributesClass MUST succeed. Reading the VolumeAttributesClass MUST
+		succeed. Patching the VolumeAttributesClass MUST succeed with its new label found. Deleting
+		the VolumeAttributesClass MUST succeed and it MUST be confirmed. Replacement VolumeAttributesClass
+		MUST be created. Updating the VolumeAttributesClass MUST succeed with its new label found.
+		Deleting the VolumeAttributesClass via deleteCollection MUST succeed and it MUST be confirmed.
+	*/
+	framework.It("should run through the lifecycle of a VolumeAttributesClass", func(ctx context.Context) {
+
+		vacClient := f.ClientSet.StorageV1alpha1().VolumeAttributesClasses()
+		var initialVAC, replacementVAC *storagev1alpha1.VolumeAttributesClass
+
+		initialVAC = &storagev1alpha1.VolumeAttributesClass{
+			TypeMeta: metav1.TypeMeta{
+				Kind: "VolumeAttributesClass",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "e2e-",
+			},
+			DriverName: "e2e-fake-csi-driver",
+			Parameters: map[string]string{
+				"foo": "bar",
+			},
+		}
+
+		ginkgo.By("Creating a VolumeAttributesClass")
+		createdVolumeAttributesClass, err := vacClient.Create(ctx, initialVAC, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create the requested VolumeAttributesClass")
+
+		ginkgo.By(fmt.Sprintf("Get VolumeAttributesClass %q", createdVolumeAttributesClass.Name))
+		retrievedVolumeAttributesClass, err := vacClient.Get(ctx, createdVolumeAttributesClass.Name, metav1.GetOptions{})
+		framework.ExpectNoError(err, "failed to get VolumeAttributesClass %q", createdVolumeAttributesClass.Name)
+
+		ginkgo.By(fmt.Sprintf("Patching the VolumeAttributesClass %q", retrievedVolumeAttributesClass.Name))
```

Comment:
Since we do not allow update the VAC do we still need this patching?

Reply:
the resource support patch and update verb and only some fields are immutable. So, I think it's still needed.